### PR TITLE
rocket: update formula to 0.3.1

### LINKED ISSUE #36490: rocket formula out of date
The latest version of rkt is [0.2.0](https://github.com/coreos/rocket/releases/tag/v0.2.0) whereas 0.1.0 is in the homebrew library. As this is pre-release software, it is going to be updated every few months.

If you could point me to the correct docs, I wouldn't mind trying to keep it up to date. However, I have no idea how to update a formula with precompiled bottles.

### PR DESCRIPTION
**Pull request by jzelinskie**
Fixes #36490 
If I need to change anything else please let me know.

### CODE REVIEW

**Comment by DomT4** [comment]
Looks like the ` inreplace ` needs another look:

```
==> go install github.com/jteeuwen/go-bindata/...
Error: inreplace failed
build:
  expected replacement of "GOOS=linux" with "GOOS=darwin\n"
```

**Comment by jzelinskie** [comment]
So, I've fixed the build script. However, `rkt` doesn't currently build on Go 1.4. Any ideas?

File: Library/Formula/rocket.rb [hunk 1]
```diff
@@ -30,12 +30,12 @@ def install
...
+    ENV["GOOS"] = `uname`.downcase.strip
 
-    # Fix non-POSIX commands realpath and mktemp
-    # See https://github.com/coreos/rocket/pull/196
-    inreplace "build", "$(mktemp -d)", "$(mktemp -d -t rocket-XXXXXX)"
-    inreplace "stage1/mkrootfs.sh", '$(realpath "${BINDIR}")', '"${PWD}${BINDIR}"'
+    # This stops [0] from executing, which is tied specifically to the Linux networking stack
+    # [0]: https://github.com/coreos/rocket/blob/6e0404bb0edc11c84a6ca5e3d0b7af9839f85cfa/build#L16-L23
+    inreplace "build", 'echo "Building network plugins"', "if false; then"
+    inreplace "build", 'echo "Building rkt (stage0)..."', "fi"
```
Comment (mikemcquaid) [inline Library/Formula/rocket.rb]: Please submit a patch upstream for this.

**Comment by mikemcquaid** [comment]
> So, I've fixed the build script. However, rkt doesn't currently build on Go 1.4 due to a concept introduced to the language called "import comments". Any ideas for building with a specific version of Go?

We'll need to just hold off on adding this until it builds with a stable Go.

File: Library/Formula/rocket.rb (hunk 1 above)
Comment (jzelinskie) [inline Library/Formula/rocket.rb]: Just submitted a PR for that.

**Comment by jzelinskie** [comment]
Some work is being done upstream to address this now.

File: Library/Formula/rocket.rb (hunk 1 above)
Comment (jzelinskie) [inline Library/Formula/rocket.rb]: This PR was merged

Comment (mikemcquaid) [inline Library/Formula/rocket.rb]: Can you link to the upstream PR and, ideally, use the commit URL as a patch URL in the formula. Thanks!

**Comment by jzelinskie** [comment]
I've made some progress, but rkt appears to be using a syscall only available on linux among other errors.

I may attempt to push forward, but rocket isn't something designed to run on OSX in the first place.
If anyone is interested on working on an OSX ACI runtime. @barakmich and I have some ideas on how to go about it.

**Comment by mikemcquaid** [comment]
> I may attempt to push forward, but rocket isn't something designed to run on OSX in the first place.

I think this is probably something you should work with upstream and create a new PR when there's a new cross-platform release. Sorry and thanks for your work here!

**Pull request closed without merging**